src/sonic-yang-models/yang-models/sonic-bgp-allowed-prefix.yang

```diff
@@ -0,0 +1,102 @@
+module sonic-bgp-allowed-prefix {
+    namespace "http://github.com/Azure/sonic-bgp-allowed-prefix";
+    prefix bgppre;
+    yang-version 1.1;
+
+    import sonic-bgp-common {
+        prefix bgpcmn;
+    }
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-routing-policy-sets {
+        prefix rpolsets;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC BGP Allowed Prefix";
+
+    revision 2022-02-26 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-bgp-allowed-prefix {
+        container BGP_ALLOWED_PREFIXES {
+            list BGP_ALLOWED_PREFIXES_COM_LIST {
+                key "deployment id community";
+
+                leaf deployment {
+                    type string {
+                        pattern "DEPLOYMENT_ID";
+                    }
+                    description "BGP allowed prefix list key type";
+                }
+
+                leaf id {
+                    type uint32;
+                    description "BGP allowed prefix list deployment id";
+                }
+
+                leaf community {
+                    type string;
```

Comment:
Does community have any pattern? 123:123?

Reply:
@ganglyu  good catch, I found there are at least 2 different formats for community id, one is 4 bytes, the other is 2 bytes:2bytes, I took a reference from existing yang model files, all of them take string as the type of community. 
I think we should put a enforcement for the community id for all cases, filed an issue to track this, address this may beyond the scope of this PR.
https://github.com/Azure/sonic-buildimage/issues/10145